src/voice_public.js

```diff
@@ -33,105 +33,116 @@ go.app = function() {
         self.states.add('state_start', function() {
             // Reset user answers when restarting the app
             self.im.user.answers = {};
+            return self.states.create("state_msg_receiver_msisdn");
 
-            return self.states.create('state_c12_number');
         });
 
-
     // CHANGE STATE
 
-        self.add('state_c12_number', function(name) {
+        // FreeText st-B
+        self.add('state_msg_receiver_msisdn', function(name) {
             var speech_option = '1';
             return new FreeText(name, {
                 question: $('Welcome, Number'),
                 helper_metadata: go.utils_project.make_voice_helper_data(
                     self.im, name, lang, speech_option),
                 next: function(content) {
-                    if (go.utils.is_valid_msisdn(content) === false) {
-                        return 'state_c13_retry_number';
+                    if (!go.utils.is_valid_msisdn(content)) {
```

Comment:
- [ ] `is_valid_msisdn` should be in `error:`, so that it replays the current state (similar to voice_registration).
- [ ] in `next:` we should be checking if the contact is registered by looking at the role_player details field, similar to what happens in `state_check_registered` in `ussd_public`.